lib/factory.rb

```diff
@@ -14,3 +16,140 @@
 # - to_a
 # - values_at
 # - ==, eql?
+# ----------------------------------------------------------------------------
+class Factory
+  def self.new(*fields, &block)
+    raise ArgumentError if fields.empty?
+
+    fields.each do |field|
+      next if field.instance_of?(Symbol)
+
+      if field.instance_of?(String) && \
+         !field.empty? && \
+         field.match(/\A[A-Z]/)
+
+        next
+      end
+
+      raise ArgumentError, "identifier #{member} needs to be a constant"
+    end
+
+    identifier, *rest = fields
+
```

Comment:
```
if constant?(identifier)
  return const_set(identifier, create_class(*rest, &block))
end

if identifier.is_a? String
  raise ArgumentError, "identifier #{identifier} needs to be a constant"
end

create_class(*fields, &block)
```

Reply:
There are no tests for validation and raising errors, do you didn't have to implement this